### PR TITLE
StepController serialization

### PR DESCRIPTION
**Pull request by wthrowe**
### Types of changes:

- [ ] Bugfix
- [ ] New feature

### Component:

- [ ] Code
- [ ] Documentation
- [ ] Build system
- [ ] Continuous integration

### Code review checklist

- [ ] The PR passes all checks, including unit tests, `clang-tidy` and `IWYU`. For
  instructions on how to perform the CI checks locally refer to the [Dev guide
  on the Travis CI](https://spectre-code.org/travis_guide.html).
- [ ] The code is documented and the documentation renders correctly. Run `make doc`
  to generate the documentation locally into `BUILD_DIR/docs/html`. Then open
  `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).

### CODE REVIEW

**Review by kidder — COMMENTED** [review]
@nilsdeppe please look at

File: src/Time/StepControllers/StepController.hpp
Comment (nilsdeppe) [inline src/Time/StepControllers/StepController.hpp]: I guess we now have to decide on whether to have any of the constructors here since the item @kidder found suggests we don't mark the destructor override since that's implicit. I'm fine saying that we only mark the base class destructor as virtual and since in this case PUP::able is the base class, we don't do anything.

File: src/Time/StepControllers/SplitRemaining.hpp
Comment (nilsdeppe) [inline src/Time/StepControllers/SplitRemaining.hpp:29]: We should probably mark all the step controllers final since I don't expect anything will inherit from them. Correct me if I'm wrong :)

Comment (kidder) [reply src/Time/StepControllers/SplitRemaining.hpp:29]: Should we put something in our coding guidelines about the use of `final`?  If so, we should have a PR discussing the use of `final` and not hold up this PR

Comment (nilsdeppe) [reply src/Time/StepControllers/SplitRemaining.hpp:29]: Use final if the class shouldn't be used as a base class... There's not more to it than that.

Comment (kidder) [reply src/Time/StepControllers/SplitRemaining.hpp:29]: Then please put in a PR or file an issue to change our coding guidelines so everyone knows what to do in the future;  I have questions but want the discussion somewhere that can be found rather than buried in this PR.  We should also add what we decided about destructors in derived classes.

**Review by nilsdeppe — DISMISSED** [review]

**Comment by nilsdeppe** [comment]
@wthrowe can you rebase this to deal with the CMakeLists conflicts?

Please don't wait for me to re-approve after the conflicts are resolved :)

**Comment by wthrowe** [comment]
Rebased.

**Review by kidder — APPROVED** [review]

**Review by mar-celine — APPROVED** [review]

**Review by osheamonn — APPROVED** [review]

**Pull request merged**
